### PR TITLE
style: formatting

### PR DESCRIPTION
**Pull request by MrAwesomeRocks**

### CODE REVIEW

File: src/gui/main_gui.cpp [hunk 1]
```diff
@@ -9,10 +9,6 @@
 #include <GLFW/glfw3.h>
 
 // Include GLM
```
Comment (AlemSnyder) [inline src/gui/main_gui.cpp:11]: if you move where GLM is included you should change the comment

File: src/main.cpp [hunk 2]
```diff
@@ -11,23 +12,23 @@
 #include <GLFW/glfw3.h>
 
 // Include GLM
-#include <glm/glm.hpp>
-#include <glm/gtc/matrix_transform.hpp>
-#include <glm/gtx/string_cast.hpp>
-
 #include "argh.h"
-
+#include "entity/mesh.hpp"
```
Comment (AlemSnyder) [inline src/main.cpp:16]: What is the include style. If you are going to change it maybe make it consistent.

File: src/terrain/chunk.cpp [hunk 3]
```diff
@@ -1,63 +1,68 @@
...
+    Cx_ = bx;
+    Cy_ = by;
+    Cz_ = bz;
+    for (int x = size * bx; x < size * (1 + bx); x++)
+        for (int y = size * by; y < size * (1 + by); y++)
+            for (int z = size * bz; z < size * (1 + bz); z++) {
+                if (ter->can_stand_1(x, y, z)) {
+                    NodeGroup group = NodeGroup(
+                        ter->get_tile(x, y, z),
+                        31); // the int determines which paths between two tiles are
```
Comment (AlemSnyder) [inline src/terrain/chunk.cpp:20]: move the comment to a different line

File: src/terrain/node_group.cpp [hunk 4]
```diff
@@ -1,84 +1,129 @@
...
-    center_z = (center_z*tiles.size() + other.center_z*other.tiles.size())/(tiles.size()+other.tiles.size());
+std::map<NodeGroup*, UnitPath>
+NodeGroup::merge_groups(NodeGroup other)
+{
+    center_x = (center_x * tiles.size() + other.center_x * other.tiles.size())
+               / (tiles.size() + other.tiles.size());
+    center_y = (center_y * tiles.size() + other.center_y * other.tiles.size())
+               / (tiles.size() + other.tiles.size());
+    center_z = (center_z * tiles.size() + other.center_z * other.tiles.size())
+               / (tiles.size() + other.tiles.size());
```
Comment (AlemSnyder) [inline src/terrain/node_group.cpp:67]: If you really care about the length of this line, change center_a to ca or something.

File: src/terrain/node_group.cpp [hunk 5]
```diff
@@ -1,84 +1,129 @@
...
+
+std::array<float, 3>
+NodeGroup::sop() const
+{
+    return {center_x, center_y, center_z};
+}
+
+const std::set<const Tile*, TilePCompare>
+NodeGroup::get_tiles() const
+{
```
Comment (AlemSnyder) [inline src/terrain/node_group.cpp:107]: Yes more lines of code this is what we need. (also inline)

File: src/terrain/terrain_generation/land_generator.cpp [hunk 6]
```diff
@@ -113,30 +127,49 @@ void LandGenerator::from_radius(int cr, int csr, TileStamp & ts) const {
...
+LandGenerator::from_grid(int cr, int csr, TileStamp& ts) const
+{
+    int x_center = (1 + 2 * (csr % data_[cr]["Grid"]["number"].asInt()))
+                       * (data_[cr]["Grid"]["radius"].asInt()
+                          / data_[cr]["Grid"]["number"].asInt())
+                   - data_[cr]["Grid"]["radius"].asInt();
+    int y_center = (1 + 2 * (csr / data_[cr]["Grid"]["number"].asInt()))
+                       * (data_[cr]["Grid"]["radius"].asInt()
+                          / data_[cr]["Grid"]["number"].asInt())
+                   - data_[cr]["Grid"]["radius"].asInt();
```
Comment (AlemSnyder) [inline src/terrain/terrain_generation/land_generator.cpp:141]: wow linting has really gotten good. this is super useful I like how this is written.

**Review by AlemSnyder — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 2 above)
Comment (MrAwesomeRocks) [reply src/main.cpp:16]: It is consistent

Vendored libs go first

File: src/gui/shadow_map.cpp [hunk 7]
```diff
@@ -102,18 +112,17 @@ void ShadowMap::render_shadow_depth_buffer() const {
...
-    //glm::mat4 depth_view_matrix =
-    //    glm::lookAt(light_direction_, glm::vec3(0, 0, 0), glm::vec3(0, 1, 0));
-    // or, for spot light :
-    // glm::vec3 lightPos(5, 20, 20);
-    // glm::mat4 depthProjectionMatrix =
-    // glm::perspective<float>(45.0f, 1.0f, 2.0f, 50.0f); glm::mat4
-    // depthViewMatrix = glm::lookAt(lightPos, lightPos-lightInvDir,
-    // glm::vec3(0,1,0));
+    // glm::mat4 depth_view_matrix =
+    //     glm::lookAt(light_direction_, glm::vec3(0, 0, 0), glm::vec3(0, 1, 0));
```
[the remark this answers is not in the document]
Comment (MrAwesomeRocks) [reply src/gui/shadow_map.cpp:113]: 2 line assignment

File: src/terrain/node_group.cpp (hunk 5 above)
Comment (MrAwesomeRocks) [reply src/terrain/node_group.cpp]: Inlining will be done later

**Comment by AlemSnyder** [comment]
lol I forgot that there is a correct way to include gl things

**Comment by AlemSnyder** [comment]
https://www.reddit.com/r/opengl/comments/1sumg2/why_does_the_order_of_my_opengl_includes_matter/

**Review by AlemSnyder — APPROVED** [review]
looks good

**Pull request merged**